src/sql/azdata.proposed.d.ts

```diff
@@ -725,6 +725,10 @@ declare module 'azdata' {
 			 * Extension can store additional information that the provider needs to uniquely identify a table.
 			 */
 			[key: string]: any;
+			/**
+			 * Table Node subtype
+			 */
+			tableSubType: TableSubType;
```

Comment:
1. rename to icon
2. this should be optional, and by default it is the table icon that we currently have.